extern/storage-sealing/checks.go

```diff
@@ -80,6 +79,24 @@ func checkPieces(ctx context.Context, maddr address.Address, si SectorInfo, api
 	return nil
 }
 
+func checkDealExpiration(ctx context.Context, sector SectorInfo, api SealingAPI) error {
```

Comment:
I don't think this is sufficiently different or faster than checkPieces to warrant its own method?

Reply:
> 1. i don't know how long `ProveReplicaUpdate1` takes, but it may be rerunning the check before calling `ProveReplicaUpdate2`?

PRU1 computes the vanilia proofs so yah it should take a while, added another check before the PRU2

Reply:
tho i do think it makes sense to check before PRU2 instead of PRU1 - updated